### ISSUE TITLE
missing_whitespace_between_adjacent_strings false positive

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/linter#2489]
In the following snippet, foo3 reports a false missing_whitespace_between_adjacent_strings. The other examples work as expected.

```dart
          String foo = 'Hello'
            ', world';

          String foo2 = 'Hello'
            '${icon != null ? ', world' : ''}';

          String foo3 = 'Hello'
            ', world' // missing_whitespace_between_adjacent_strings incorrectly reported
            '${icon != null ? ', world' : ''}';
```

**Comment by goderbauer** [comment]
Actual code in the flutter framework that triggered the false positive:

https://github.com/flutter/flutter/blob/3594d1aa47ffa37df1fda45d861a1d4f68af840b/packages/flutter/test/material/text_button_test.dart#L834-L837

**Comment by a14n** [comment]
FWIW the code could perhaps read better with list-join:

```dart
final String testName = [
  'TextButton',
  'text scale $textScaleFactor',
  if (icon != null) 'with icon',
  if (textDirection == TextDirection.rtl) 'RTL',
].join(', ');
```
